spinnaker-monitoring-daemon/spinnaker-monitoring/spectator_metric_transformer.py

```diff
@@ -25,15 +25,23 @@
 import sys
 import yaml
 
-
-# see https://stackoverflow.com/questions/1175208/\
-#     elegant-python-function-to-convert-camelcase-to-snake-case
-_SNAKE_CASE_RE = re.compile('((?<=[a-z0-9])[A-Z]|(?!^)(?<!_)[A-Z](?=[a-z]))')
-
+# This is a cache for the _snakeify function to increase performance on the
+#    conversion of CamelCase to snake_case
+SNAKEIFY_CACHE = {}
 
 def _snakeify(text):
-  """Turn text into snake-case."""
-  return _SNAKE_CASE_RE.sub(r'_\1', text).lower()
+  if text not in SNAKEIFY_CACHE:
+    result = []
+    result.append(text[0].lower())
+    for position in range(1, len(text)):
+      if text[position].isupper():
+        if text[position - 1].islower():
+          result.append('_')
+        elif position < (len(text) - 1) and text[position + 1].islower() and text[position - 1] is not '_':
```

Comment:
Maybe a micro-optimization...but we're doing `len(text)` on every iteration; could this be a local variable?

Reply:
Added this micro-optimization.